YAKL_defines.h

```diff
@@ -16,8 +16,8 @@
   #include <hipcub/hipcub.hpp>
 #elif defined(__USE_SYCL__)
   #define YAKL_LAMBDA [=]
-  #define YAKL_INLINE inline
-  #define YAKL_DEVICE inline
+  #define YAKL_INLINE __inline__ __attribute__((always_inline))
```

Comment:
This is Clang-specific. Won't work on GCC. We're only doing SYCL for Intel anyway.